File: katas/7 kyu/find-the-middle-element/solution.py
# find-the-middle-element
import codewars_test as test
# -------------------------------- MY SOLUTION ------------------------------- #
def gimme(input_array):
  for i,x in enumerate(input_array): 
    if x != max(input_array) and x != min(input_array): return i
# ------------------------------ CLEVER SOLUTION ----------------------------- #
# ----------------------------------- TEST ----------------------------------- #
test.assert_equals(gimme([2, 3, 1]), 0, 'Finds the index of middle element')
test.assert_equals(gimme([5, 10, 14]), 1, 'Finds the index of middle element')

File: katas/7 kyu/find-the-middle-element/solution.js
// find-the-middle-element
/* ------------------------------- MY SOLUTION ------------------------------ */
function gimme (triplet) {
  for (const [index, element] of triplet.entries()) {
    if (element != Math.max(...triplet) && element != Math.min(...triplet)) return index
  }
}
/* ----------------------------- CLEVER SOLUTION ---------------------------- */
/* ---------------------------------- TEST ---------------------------------- */
// const chai = require("chai");
// const Test = chai.assert.strictEqual; // replace Test.assertEqual with Test
// chai.config.truncateThreshold=0;

const strictEqual = require('chai').assert.strictEqual;

function doTest (triplet, expected) {
	const actual = gimme(triplet);
	strictEqual(actual, expected, `for [${triplet}], expected ${expected} but got ${actual}`);
}

describe("Basic Test", function () {
	it("Tests for integers", function () {
		doTest([2, 3, 1], 0);
		doTest([5, 10, 14], 1);
	});
	it("Tests for floats", function () {
		doTest([2.1, 3.2, 1.4], 0);
		doTest([5.9, 10.4, 14.2], 1);
	});
	it("Tests for negative numbers", function () {
		doTest([-2, -3, -1], 0);
		doTest([-5, -10, -14], 1);
	});
	it("Tests for mixed numbers", function () {
		doTest([-2, -3.2, 1], 0);
		doTest([-5.2, -10.6, 14], 0);
	});
});